tests/test_chmod_command.rs

```diff
@@ -0,0 +1,191 @@
+    files: &[(&str, &str)],
+) {
+    if parents.is_empty() {
+        test_env.jj_cmd_success(repo_path, &["new", "root", "-m", name]);
+    } else {
+        let mut args = vec!["new", "-m", name];
+        args.extend(parents);
+        test_env.jj_cmd_success(repo_path, &args);
+    }
+    for (name, content) in files {
+        std::fs::write(repo_path.join(name), content).unwrap();
+    }
+    test_env.jj_cmd_success(repo_path, &["branch", "create", name]);
+}
+
+fn get_log_output(test_env: &TestEnvironment, repo_path: &Path) -> String {
+    test_env.jj_cmd_success(repo_path, &["log", "-T", "branches"])
+}
+
+#[test]
+fn test_chmod_regular_conflict() {
+    let test_env = TestEnvironment::default();
+    test_env.jj_cmd_success(test_env.env_root(), &["init", "repo", "--git"]);
+    let repo_path = test_env.env_root().join("repo");
+
+    create_commit(&test_env, &repo_path, "base", &[], &[("file", "base\n")]);
+    create_commit(&test_env, &repo_path, "n", &["base"], &[("file", "n\n")]);
+    create_commit(&test_env, &repo_path, "x", &["base"], &[("file", "x\n")]);
+    // Test chmodding a file. The effect will be visible in the conflict below.
+    test_env.jj_cmd_success(&repo_path, &["chmod", "x", "file", "-r=x"]);
+    create_commit(&test_env, &repo_path, "conflict", &["x", "n"], &[]);
+
+    // Test the setup
+    insta::assert_snapshot!(get_log_output(&test_env, &repo_path), @r###"
+    @    conflict
+    ├─╮
+    ◉ │  x
+    │ ◉  n
+    ├─╯
+    ◉  base
+    ◉
+    "###);
+    let stdout = test_env.jj_cmd_success(&repo_path, &["cat", "file"]);
+    insta::assert_snapshot!(stdout, 
+    @r###"
+    Conflict:
+      Removing file with id df967b96a579e45a18b8251732d16804b2e56a55
+      Adding executable file with id 587be6b4c3f93f93c489c0111bba5596147a26cb
+      Adding file with id 8ba3a16384aacc37d01564b28401755ce8053f51
+    "###);
+
+    // Test chmodding a conflict
+    test_env.jj_cmd_success(&repo_path, &["chmod", "x", "file"]);
+    let stdout = test_env.jj_cmd_success(&repo_path, &["cat", "file"]);
+    insta::assert_snapshot!(stdout, 
+    @r###"
+    Conflict:
+      Removing executable file with id df967b96a579e45a18b8251732d16804b2e56a55
+      Adding executable file with id 587be6b4c3f93f93c489c0111bba5596147a26cb
+      Adding executable file with id 8ba3a16384aacc37d01564b28401755ce8053f51
+    "###);
+    test_env.jj_cmd_success(&repo_path, &["chmod", "n", "file"]);
+    let stdout = test_env.jj_cmd_success(&repo_path, &["cat", "file"]);
+    insta::assert_snapshot!(stdout, 
+    @r###"
+    <<<<<<<
+    %%%%%%%
+    -base
+    +x
+    +++++++
+    n
+    >>>>>>>
+    "###);
+
+    // An error prevents `chmod` from making any changes.
+    // In this case, the failure with `nonexistent` prevents any changes to `file`.
+    let stderr = test_env.jj_cmd_failure(&repo_path, &["chmod", "x", "nonexistent", "file"]);
+    insta::assert_snapshot!(stderr, @r###"
+    Error: Can't find anything at `nonexistent`.
```

Comment:
`jj cat` says "No such path" (without mentioning which one, because it only supports one path). So maybe "No such path: <path>" here?

We seem to mostly use double and single quotes in error message. We should probably standardize on one type of quotes. I don't have much opinion about which one (single, double, back), but I would suggest using double or single for now unless you feel like updating the other places to also use backticks.

Reply:
I changed it. I like having the `at` in the message, which doesn't work quite as well with "No such path", but I think it's OK.